exporters/otlp/otlpmetric/otlpmetrichttp/config.go

```diff
@@ -149,6 +149,12 @@ func WithHeaders(headers map[string]string) Option {
 	return wrappedOption{oconf.WithHeaders(headers)}
 }
 
+// WithHostHeader overrides Host header in HTTP requests.
+// If unset, the target Endpoint's host will be used as Host header.
```

Comment:
How about a more detailed doc here, with an explanation of the Envoy case?

Reply:
ok, I will add that. What do you think about adding environment variable for this? I thought to add but didn't because wasn't sure that MR will be approved this way with a separate WithHostHeaderOption

Reply:
I'd say let's do it in another PR to keep things more focused.